### DISCUSSION TITLE
treat string as a custom (uuid) object

### DISCUSSION THREAD

**Discussion by antony-amdl** [rrousselGit/freezed#805] [Q&A]
I have
```dart
import 'package:uuid/uuid.dart' as uuid_lib;

const uuidGen = uuid_lib.Uuid();

class Uuid {
  final String _val;
  Uuid._private(this._val);

  factory Uuid() => Uuid._private(uuidGen.v4());

  factory Uuid.fromJson(String json) => Uuid._private(json);

  String toJson() => _val;
}
```
This is very similar to `sample2` in https://pub.dev/packages/json_serializable#custom-types-and-custom-encoding  
I try to use it in
```dart
import 'package:flutter/foundation.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

import '../util/uuid.dart';

part 'organization.freezed.dart';
part 'organization.g.dart';

@freezed
class Organization with _$Organization {
  const factory Organization({
    required Uuid pkid,
    required String name,
  }) = _Organization;

  factory Organization.fromJson(Map<String, dynamic> json) =>
      _$OrganizationFromJson(json);
}
```
The idea is to send and receive uuid fields as just string, but treat it as an opaque object throughout the flutter code.
But this does not work since the generated code for the container class `Organization`  does not call the `toJson` method on the `Uuid` type field.
(If I find a way to do this, I plan to apply similar methods on iso-8601 timestamps, email, phone etc)
Please help.

**Comment by TekExplorer** [comment]
It looks like it does call tojson and fromjson just fine.

**Comment by antony-amdl** [comment]
please see https://discord.com/channels/765557403865186374/860193850029768724/1054602120642441216 for reproducible test case
